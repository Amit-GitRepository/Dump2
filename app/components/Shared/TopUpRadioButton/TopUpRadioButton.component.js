import noop from 'lodash/noop';
import Proptypes from 'prop-types';
import React, {Component} from 'react';
import styles from './TopUpRadioButton.style';
import {ISText, Touchable} from '../../Shared';
import {Platform, View} from 'react-native';

class TopUpRadioButton extends Component {
  onToggle = () => {
    const {radioSelect, value} = this.props;
    radioSelect(value);
  }
  render () {
    const {active, label, style} = this.props;
    const activeContainerStyles = active ? styles.active : styles.inactive;
    const inactiveText = (Platform.OS === 'android') ? [styles.inactiveText, styles.androidText] : styles.inactiveText;
    const activeTextStyles = active ? styles.activeText : inactiveText;
    return (
      <Touchable borderless={true} onPress={this.onToggle} style={styles.clickableBox}>
        <View style={[styles.container, activeContainerStyles, style]}>
          <ISText type={active ? 'BOLD' : 'MEDIUM'} style={[styles.text, activeTextStyles]}>{label}</ISText>
        </View>
      </Touchable>);
  }
}
TopUpRadioButton.defaultProps = {
  active: false,
  label: '',
  value: '',
  radioSelect: noop,
  style: {}
};
TopUpRadioButton.propTypes = {
  active: Proptypes.bool,
  label: Proptypes.string,
  value: Proptypes.string,
  radioSelect: Proptypes.func,
  style: Proptypes.object
};
export default TopUpRadioButton;
