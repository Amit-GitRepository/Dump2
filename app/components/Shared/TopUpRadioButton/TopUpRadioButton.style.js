import {colors, fonts} from '../../../themes/constants.styles';

export default {
  clickableBox: {
    flex: 1,
    borderRadius: 5
  },
  container: {
    flex: 0,
    borderWidth: 2,
    borderColor: colors.PRIMARY_ACTIONABLE,
    borderRadius: 5,
    alignItems: 'center',
    justifyContent: 'center',
    marginRight: 5
  },
  active: {
    backgroundColor: colors.PRIMARY_ACTIONABLE
  },
  inactive: {
    backgroundColor: colors.TRANSPARENT
  },
  text: {
    fontSize: fonts.FONT_SIZE_NORMAL,
    alignSelf: 'center'
  },
  activeText: {
    color: colors.PRIMARY_BG_TEXT_CONTRAST
  },
  inactiveText: {
    color: colors.PRIMARY_TEXT_TAB_LABEL
  },
  androidText: {
    paddingVertical: 1
  }
};
