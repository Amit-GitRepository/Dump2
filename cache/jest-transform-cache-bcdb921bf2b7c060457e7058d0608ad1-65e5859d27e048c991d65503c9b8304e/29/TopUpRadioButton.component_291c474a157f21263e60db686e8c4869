Object.defineProperty(exports,"__esModule",{value:true});var cov_284h18gy19=function(){var path='/Users/amitkumar/iTrueService/DevCode/iservice-app/app/components/BillUsage/BillUsage/BillUsageShared/TopUpRadioButton.component.js',hash='36ffaaa27c2f4c30da0f4bf8d59230a8be8ad423',global=new Function('return this')(),gcv='__coverage__',coverageData={path:'/Users/amitkumar/iTrueService/DevCode/iservice-app/app/components/BillUsage/BillUsage/BillUsageShared/TopUpRadioButton.component.js',statementMap:{'0':{start:{line:11,column:33},end:{line:11,column:43}},'1':{start:{line:12,column:4},end:{line:12,column:23}},'2':{start:{line:15,column:28},end:{line:15,column:38}},'3':{start:{line:16,column:34},end:{line:16,column:74}},'4':{start:{line:17,column:29},end:{line:17,column:77}},'5':{start:{line:18,column:4},end:{line:23,column:20}},'6':{start:{line:26,column:0},end:{line:31,column:2}},'7':{start:{line:32,column:0},end:{line:37,column:2}}},fnMap:{'0':{name:'(anonymous_0)',decl:{start:{line:10,column:13},end:{line:10,column:14}},loc:{start:{line:10,column:19},end:{line:13,column:3}},line:10},'1':{name:'(anonymous_1)',decl:{start:{line:14,column:2},end:{line:14,column:3}},loc:{start:{line:14,column:12},end:{line:24,column:3}},line:14}},branchMap:{'0':{loc:{start:{line:16,column:34},end:{line:16,column:74}},type:'cond-expr',locations:[{start:{line:16,column:43},end:{line:16,column:56}},{start:{line:16,column:59},end:{line:16,column:74}}],line:16},'1':{loc:{start:{line:17,column:29},end:{line:17,column:77}},type:'cond-expr',locations:[{start:{line:17,column:38},end:{line:17,column:55}},{start:{line:17,column:58},end:{line:17,column:77}}],line:17},'2':{loc:{start:{line:21,column:24},end:{line:21,column:50}},type:'cond-expr',locations:[{start:{line:21,column:33},end:{line:21,column:39}},{start:{line:21,column:42},end:{line:21,column:50}}],line:21}},s:{'0':0,'1':0,'2':0,'3':0,'4':0,'5':0,'6':0,'7':0},f:{'0':0,'1':0},b:{'0':[0,0],'1':[0,0],'2':[0,0]},_coverageSchema:'332fd63041d2c1bcb487cc26dd0d5f7d97098a6c'},coverage=global[gcv]||(global[gcv]={});if(coverage[path]&&coverage[path].hash===hash){return coverage[path];}coverageData.hash=hash;return coverage[path]=coverageData;}();var _createClass=function(){function defineProperties(target,props){for(var i=0;i<props.length;i++){var descriptor=props[i];descriptor.enumerable=descriptor.enumerable||false;descriptor.configurable=true;if("value"in descriptor)descriptor.writable=true;Object.defineProperty(target,descriptor.key,descriptor);}}return function(Constructor,protoProps,staticProps){if(protoProps)defineProperties(Constructor.prototype,protoProps);if(staticProps)defineProperties(Constructor,staticProps);return Constructor;};}();var _ISText=require('../../../Shared/ISText/ISText.component');var _ISText2=_interopRequireDefault(_ISText);
var _noop=require('lodash/noop');var _noop2=_interopRequireDefault(_noop);
var _propTypes=require('prop-types');var _propTypes2=_interopRequireDefault(_propTypes);
var _react=require('react');var _react2=_interopRequireDefault(_react);
var _TopUpRadioButton=require('./TopUpRadioButton.style');var _TopUpRadioButton2=_interopRequireDefault(_TopUpRadioButton);
var _Touchable=require('../../../Shared/Touchable/Touchable.component');var _Touchable2=_interopRequireDefault(_Touchable);
var _reactNative=require('react-native');function _interopRequireDefault(obj){return obj&&obj.__esModule?obj:{default:obj};}function _classCallCheck(instance,Constructor){if(!(instance instanceof Constructor)){throw new TypeError("Cannot call a class as a function");}}function _possibleConstructorReturn(self,call){if(!self){throw new ReferenceError("this hasn't been initialised - super() hasn't been called");}return call&&(typeof call==="object"||typeof call==="function")?call:self;}function _inherits(subClass,superClass){if(typeof superClass!=="function"&&superClass!==null){throw new TypeError("Super expression must either be null or a function, not "+typeof superClass);}subClass.prototype=Object.create(superClass&&superClass.prototype,{constructor:{value:subClass,enumerable:false,writable:true,configurable:true}});if(superClass)Object.setPrototypeOf?Object.setPrototypeOf(subClass,superClass):subClass.__proto__=superClass;}var

TopUpRadioButton=function(_Component){_inherits(TopUpRadioButton,_Component);function TopUpRadioButton(){var _ref;var _temp,_this,_ret;_classCallCheck(this,TopUpRadioButton);for(var _len=arguments.length,args=Array(_len),_key=0;_key<_len;_key++){args[_key]=arguments[_key];}return _ret=(_temp=(_this=_possibleConstructorReturn(this,(_ref=TopUpRadioButton.__proto__||Object.getPrototypeOf(TopUpRadioButton)).call.apply(_ref,[this].concat(args))),_this),_this.
onToggle=function(){cov_284h18gy19.f[0]++;var _ref2=(cov_284h18gy19.s[0]++,
_this.props),radioSelect=_ref2.radioSelect,value=_ref2.value;cov_284h18gy19.s[1]++;
radioSelect(value);
},_temp),_possibleConstructorReturn(_this,_ret);}_createClass(TopUpRadioButton,[{key:'render',value:function render()
{cov_284h18gy19.f[1]++;var _ref3=(cov_284h18gy19.s[2]++,
this.props),active=_ref3.active,label=_ref3.label;
var activeContainerStyles=(cov_284h18gy19.s[3]++,active?(cov_284h18gy19.b[0][0]++,_TopUpRadioButton2.default.active):(cov_284h18gy19.b[0][1]++,_TopUpRadioButton2.default.inactive));
var activeTextStyles=(cov_284h18gy19.s[4]++,active?(cov_284h18gy19.b[1][0]++,_TopUpRadioButton2.default.activeText):(cov_284h18gy19.b[1][1]++,_TopUpRadioButton2.default.inactiveText));cov_284h18gy19.s[5]++;
return(
_react2.default.createElement(_Touchable2.default,{onPress:this.onToggle,style:_TopUpRadioButton2.default.clickableBox},
_react2.default.createElement(_reactNative.View,{style:[_TopUpRadioButton2.default.container,activeContainerStyles]},
_react2.default.createElement(_ISText2.default,{type:active?(cov_284h18gy19.b[2][0]++,'BOLD'):(cov_284h18gy19.b[2][1]++,'MEDIUM'),style:[_TopUpRadioButton2.default.text,activeTextStyles]},label))));


}}]);return TopUpRadioButton;}(_react.Component);cov_284h18gy19.s[6]++;

TopUpRadioButton.defaultProps={
active:false,
label:'',
value:'',
radioSelect:_noop2.default};cov_284h18gy19.s[7]++;

TopUpRadioButton.propTypes={
active:_propTypes2.default.bool,
label:_propTypes2.default.string,
value:_propTypes2.default.string,
radioSelect:_propTypes2.default.func};exports.default=

TopUpRadioButton;