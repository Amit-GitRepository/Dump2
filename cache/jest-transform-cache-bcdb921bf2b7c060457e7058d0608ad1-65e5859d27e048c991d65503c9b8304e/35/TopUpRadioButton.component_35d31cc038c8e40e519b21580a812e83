Object.defineProperty(exports,"__esModule",{value:true});var cov_22qubxroka=function(){var path='/Users/amitkumar/iTrueService/DevCode/iservice-app/app/components/Shared/TopUpRadioButton/TopUpRadioButton.component.js',hash='3d8653410e57834c23094cf5893a918ecd9c9a9c',global=new Function('return this')(),gcv='__coverage__',coverageData={path:'/Users/amitkumar/iTrueService/DevCode/iservice-app/app/components/Shared/TopUpRadioButton/TopUpRadioButton.component.js',statementMap:{'0':{start:{line:10,column:33},end:{line:10,column:43}},'1':{start:{line:11,column:4},end:{line:11,column:23}},'2':{start:{line:14,column:35},end:{line:14,column:45}},'3':{start:{line:15,column:34},end:{line:15,column:74}},'4':{start:{line:16,column:29},end:{line:16,column:77}},'5':{start:{line:17,column:4},end:{line:22,column:20}},'6':{start:{line:25,column:0},end:{line:31,column:2}},'7':{start:{line:32,column:0},end:{line:38,column:2}}},fnMap:{'0':{name:'(anonymous_0)',decl:{start:{line:9,column:13},end:{line:9,column:14}},loc:{start:{line:9,column:19},end:{line:12,column:3}},line:9},'1':{name:'(anonymous_1)',decl:{start:{line:13,column:2},end:{line:13,column:3}},loc:{start:{line:13,column:12},end:{line:23,column:3}},line:13}},branchMap:{'0':{loc:{start:{line:15,column:34},end:{line:15,column:74}},type:'cond-expr',locations:[{start:{line:15,column:43},end:{line:15,column:56}},{start:{line:15,column:59},end:{line:15,column:74}}],line:15},'1':{loc:{start:{line:16,column:29},end:{line:16,column:77}},type:'cond-expr',locations:[{start:{line:16,column:38},end:{line:16,column:55}},{start:{line:16,column:58},end:{line:16,column:77}}],line:16},'2':{loc:{start:{line:20,column:24},end:{line:20,column:50}},type:'cond-expr',locations:[{start:{line:20,column:33},end:{line:20,column:39}},{start:{line:20,column:42},end:{line:20,column:50}}],line:20}},s:{'0':0,'1':0,'2':0,'3':0,'4':0,'5':0,'6':0,'7':0},f:{'0':0,'1':0},b:{'0':[0,0],'1':[0,0],'2':[0,0]},_coverageSchema:'332fd63041d2c1bcb487cc26dd0d5f7d97098a6c'},coverage=global[gcv]||(global[gcv]={});if(coverage[path]&&coverage[path].hash===hash){return coverage[path];}coverageData.hash=hash;return coverage[path]=coverageData;}();var _createClass=function(){function defineProperties(target,props){for(var i=0;i<props.length;i++){var descriptor=props[i];descriptor.enumerable=descriptor.enumerable||false;descriptor.configurable=true;if("value"in descriptor)descriptor.writable=true;Object.defineProperty(target,descriptor.key,descriptor);}}return function(Constructor,protoProps,staticProps){if(protoProps)defineProperties(Constructor.prototype,protoProps);if(staticProps)defineProperties(Constructor,staticProps);return Constructor;};}();var _noop=require('lodash/noop');var _noop2=_interopRequireDefault(_noop);
var _propTypes=require('prop-types');var _propTypes2=_interopRequireDefault(_propTypes);
var _react=require('react');var _react2=_interopRequireDefault(_react);
var _TopUpRadioButton=require('./TopUpRadioButton.style');var _TopUpRadioButton2=_interopRequireDefault(_TopUpRadioButton);
var _Shared=require('../../Shared');
var _reactNative=require('react-native');function _interopRequireDefault(obj){return obj&&obj.__esModule?obj:{default:obj};}function _classCallCheck(instance,Constructor){if(!(instance instanceof Constructor)){throw new TypeError("Cannot call a class as a function");}}function _possibleConstructorReturn(self,call){if(!self){throw new ReferenceError("this hasn't been initialised - super() hasn't been called");}return call&&(typeof call==="object"||typeof call==="function")?call:self;}function _inherits(subClass,superClass){if(typeof superClass!=="function"&&superClass!==null){throw new TypeError("Super expression must either be null or a function, not "+typeof superClass);}subClass.prototype=Object.create(superClass&&superClass.prototype,{constructor:{value:subClass,enumerable:false,writable:true,configurable:true}});if(superClass)Object.setPrototypeOf?Object.setPrototypeOf(subClass,superClass):subClass.__proto__=superClass;}var

TopUpRadioButton=function(_Component){_inherits(TopUpRadioButton,_Component);function TopUpRadioButton(){var _ref;var _temp,_this,_ret;_classCallCheck(this,TopUpRadioButton);for(var _len=arguments.length,args=Array(_len),_key=0;_key<_len;_key++){args[_key]=arguments[_key];}return _ret=(_temp=(_this=_possibleConstructorReturn(this,(_ref=TopUpRadioButton.__proto__||Object.getPrototypeOf(TopUpRadioButton)).call.apply(_ref,[this].concat(args))),_this),_this.
onToggle=function(){cov_22qubxroka.f[0]++;var _ref2=(cov_22qubxroka.s[0]++,
_this.props),radioSelect=_ref2.radioSelect,value=_ref2.value;cov_22qubxroka.s[1]++;
radioSelect(value);
},_temp),_possibleConstructorReturn(_this,_ret);}_createClass(TopUpRadioButton,[{key:'render',value:function render()
{cov_22qubxroka.f[1]++;var _ref3=(cov_22qubxroka.s[2]++,
this.props),active=_ref3.active,label=_ref3.label,style=_ref3.style;
var activeContainerStyles=(cov_22qubxroka.s[3]++,active?(cov_22qubxroka.b[0][0]++,_TopUpRadioButton2.default.active):(cov_22qubxroka.b[0][1]++,_TopUpRadioButton2.default.inactive));
var activeTextStyles=(cov_22qubxroka.s[4]++,active?(cov_22qubxroka.b[1][0]++,_TopUpRadioButton2.default.activeText):(cov_22qubxroka.b[1][1]++,_TopUpRadioButton2.default.inactiveText));cov_22qubxroka.s[5]++;
return(
_react2.default.createElement(_Shared.Touchable,{onPress:this.onToggle,style:_TopUpRadioButton2.default.clickableBox},
_react2.default.createElement(_reactNative.View,{style:[_TopUpRadioButton2.default.container,activeContainerStyles,style]},
_react2.default.createElement(_Shared.ISText,{type:active?(cov_22qubxroka.b[2][0]++,'BOLD'):(cov_22qubxroka.b[2][1]++,'MEDIUM'),style:[_TopUpRadioButton2.default.text,activeTextStyles]},label))));


}}]);return TopUpRadioButton;}(_react.Component);cov_22qubxroka.s[6]++;

TopUpRadioButton.defaultProps={
active:false,
label:'',
value:'',
radioSelect:_noop2.default,
style:{}};cov_22qubxroka.s[7]++;

TopUpRadioButton.propTypes={
active:_propTypes2.default.bool,
label:_propTypes2.default.string,
value:_propTypes2.default.string,
radioSelect:_propTypes2.default.func,
style:_propTypes2.default.object};exports.default=

TopUpRadioButton;