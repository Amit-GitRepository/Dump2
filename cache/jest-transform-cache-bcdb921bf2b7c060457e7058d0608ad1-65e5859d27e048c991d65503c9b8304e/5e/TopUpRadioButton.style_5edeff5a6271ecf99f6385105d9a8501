Object.defineProperty(exports,"__esModule",{value:true});var cov_puvres8v2=function(){var path='/Users/amitkumar/iTrueService/DevCode/iservice-app/app/components/Shared/TopUpRadioButton/TopUpRadioButton.style.js',hash='1dfe6c4623ad28dccde08126f0d92d6571c88d5a',global=new Function('return this')(),gcv='__coverage__',coverageData={path:'/Users/amitkumar/iTrueService/DevCode/iservice-app/app/components/Shared/TopUpRadioButton/TopUpRadioButton.style.js',statementMap:{},fnMap:{},branchMap:{},s:{},f:{},b:{},_coverageSchema:'332fd63041d2c1bcb487cc26dd0d5f7d97098a6c'},coverage=global[gcv]||(global[gcv]={});if(coverage[path]&&coverage[path].hash===hash){return coverage[path];}coverageData.hash=hash;return coverage[path]=coverageData;}();var _constants=require('../../../themes/constants.styles');exports.default=

{
clickableBox:{
flexGrow:1},

container:{
flex:0,
borderWidth:2,
borderColor:_constants.colors.PRIMARY_ACTIONABLE,
borderRadius:5,
alignItems:'center',
justifyContent:'center',
marginRight:5},

active:{
backgroundColor:_constants.colors.PRIMARY_ACTIONABLE},

inactive:{
backgroundColor:_constants.colors.TRANSPARENT},

text:{
fontSize:_constants.fonts.FONT_SIZE_NORMAL,
alignSelf:'center'},

activeText:{
color:_constants.colors.PRIMARY_BG_TEXT_CONTRAST},

inactiveText:{
color:_constants.colors.PRIMARY_TEXT_TAB_LABEL}};