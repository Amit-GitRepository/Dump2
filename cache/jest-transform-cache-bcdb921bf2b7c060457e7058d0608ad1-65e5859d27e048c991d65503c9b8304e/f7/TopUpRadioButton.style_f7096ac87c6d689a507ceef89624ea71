Object.defineProperty(exports,"__esModule",{value:true});var cov_2p06jqmcx1=function(){var path='/Users/amitkumar/iTrueService/DevCode/iservice-app/app/components/BillUsage/BillUsage/BillUsageShared/TopUpRadioButton.style.js',hash='509ecd819d8e547317872d05d3925c2831d8ff04',global=new Function('return this')(),gcv='__coverage__',coverageData={path:'/Users/amitkumar/iTrueService/DevCode/iservice-app/app/components/BillUsage/BillUsage/BillUsageShared/TopUpRadioButton.style.js',statementMap:{},fnMap:{},branchMap:{},s:{},f:{},b:{},_coverageSchema:'332fd63041d2c1bcb487cc26dd0d5f7d97098a6c'},coverage=global[gcv]||(global[gcv]={});if(coverage[path]&&coverage[path].hash===hash){return coverage[path];}coverageData.hash=hash;return coverage[path]=coverageData;}();var _constants=require('../../../../themes/constants.styles');exports.default=

{
clickableBox:{
flex:1},

container:{
borderWidth:2,
borderColor:_constants.colors.PRIMARY_ACTIONABLE,
borderRadius:5,
alignItems:'center',
justifyContent:'center',
marginRight:5},

active:{
backgroundColor:_constants.colors.PRIMARY_ACTIONABLE},

inactive:{
backgroundColor:_constants.colors.TRANSPARENT},

text:{
fontSize:_constants.fonts.FONT_SIZE_NORMAL,
alignSelf:'center'},

activeText:{
color:_constants.colors.PRIMARY_BG_TEXT_CONTRAST},

inactiveText:{
color:_constants.colors.PRIMARY_TEXT_TAB_LABEL}};