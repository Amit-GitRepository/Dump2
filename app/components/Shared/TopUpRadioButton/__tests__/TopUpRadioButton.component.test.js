import React from 'react';
import ReactSixteenAdapter from 'enzyme/build/adapters/ReactSixteenAdapter';
import renderer from 'react-test-renderer';
import TopUpRadioButton from '../TopUpRadioButton.component';
import {configure, shallow} from 'enzyme';

configure({adapter: new ReactSixteenAdapter()});

describe('TopUpRadioButton component', () => {
  it('renders correctly', () => {
    const component = renderer.create(
      <TopUpRadioButton label='First' value='1' index={0} radioSelect={jest.fn()}/>).toJSON();
    expect(component).toMatchSnapshot();
  });
  it('renders correctly Radio button with the props', () => {
    const wrapper = shallow(<TopUpRadioButton label='First' value='1' index={0} radioSelect={jest.fn()}/>);
    expect(wrapper).toBeDefined();
  });
  it('onToggle should callback to radioSelect func', () => {
    const wrapper = shallow(<TopUpRadioButton label='First' value='1' index={0} radioSelect={jest.fn()}/>);
    wrapper.instance().onToggle();
    expect(wrapper.instance().props.radioSelect).toHaveBeenCalled();
  });
  it('onToggle should callback to radioSelect func with value', () => {
    const wrapper = shallow(<TopUpRadioButton label='First' value='1' index={0} radioSelect={jest.fn()}/>);
    wrapper.instance().onToggle();
    const props = wrapper.instance().props;
    expect(wrapper.instance().props.radioSelect).toHaveBeenCalled();
    expect(props.radioSelect).toHaveBeenLastCalledWith(props.value);
  });
});
