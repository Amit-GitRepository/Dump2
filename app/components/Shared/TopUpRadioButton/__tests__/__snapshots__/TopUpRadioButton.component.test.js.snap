// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`TopUpRadioButton component renders correctly 1`] = `
<View
  accessibilityComponentType={undefined}
  accessibilityLabel={undefined}
  accessibilityTraits={undefined}
  accessible={true}
  collapsable={undefined}
  hitSlop={undefined}
  isTVSelectable={true}
  nativeID={undefined}
  onLayout={undefined}
  onResponderGrant={[Function]}
  onResponderMove={[Function]}
  onResponderRelease={[Function]}
  onResponderTerminate={[Function]}
  onResponderTerminationRequest={[Function]}
  onStartShouldSetResponder={[Function]}
  style={
    Object {
      "borderRadius": 5,
      "flex": 1,
      "opacity": 1,
    }
  }
  testID={undefined}
  tvParallaxProperties={undefined}
>
  <View
    style={
      Object {
        "opacity": 1,
      }
    }
  >
    <View
      style={
        Array [
          Object {
            "alignItems": "center",
            "borderColor": "#F93E60",
            "borderRadius": 5,
            "borderWidth": 2,
            "flex": 0,
            "justifyContent": "center",
            "marginRight": 5,
          },
          Object {
            "backgroundColor": "transparent",
          },
          Object {},
        ]
      }
    >
      <Text
        accessible={true}
        allowFontScaling={true}
        ellipsizeMode="tail"
        style={
          Array [
            Object {
              "backgroundColor": "transparent",
              "color": "#303C46",
            },
            Array [
              Object {
                "alignSelf": "center",
                "fontSize": 16,
              },
              Object {
                "color": "#303C46",
              },
            ],
            Object {
              "fontFamily": "SukhumvitSet-Medium",
            },
          ]
        }
      >
        First
      </Text>
    </View>
  </View>
</View>
`;
